specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/assessments.json

```diff
@@ -0,0 +1,250 @@
+      "type": "object",
+      "description": "Security assessment on a resource",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/SecurityAssessmentProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
+        }
+      ]
+    },
+    "SecurityAssessmentProperties": {
+      "type": "object",
+      "description": "Describes properties of an assessment.",
+      "properties": {
+        "resourceDetails": {
+          "$ref": "#/definitions/ResourceDetails"
+        },
+        "displayName": {
+          "readOnly": true,
+          "type": "string",
+          "description": "User friendly display name of the assessment"
+        },
+        "status": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Status of the ",
+          "enum": ["Passed", "Failed", "NotApplicable"],
+          "x-ms-enum": {
+            "name": "Severity",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Passed",
+                "description": "The resource is healthy"
+              },
+              {
+                "value": "Failed",
+                "description": "The resource has a security issue that needs to be addressed"
+              },
+              {
+                "value": "NotApplicable",
+                "description": "Assessment for this resource did not happen"
+              }
+            ]
+          }
+        },
+        "statusReasonCode": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Programmatic code for the reason the assessment result status"
+        },
+        "statusReasonDescription": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Human readable description for the reason the assessment result status"
+        },
+        "additionalData": {
+          "readOnly": true,
+          "type": "array",
+          "description": "Additional data regarding the assessment",
+          "items":{
+            "$ref": "#/definitions/AssessmentAdditionalData"
+          }
+        }
+      }
+    },
+    "AssessmentAdditionalData": {
+      "type": "object",
+      "description": "Additional data regarding the assessment",
+      "properties": {
+        "key": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Name of the field of the additional data"
+        },
+        "value": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Value of the field of the additional data"
+        }
+      }
+    },
+    "ResourceDetails": {
+      "type": "object",
+      "description": "Details of the resource that was assessed",
+      "discriminator": "assessedResourceSource",
+      "properties": {
+        "assessedResourceSource": {

```

Comment:
>assessedResourceSource [](start = 9, length = 22)

This should probably be an enum that only allows "Azure" as the value if you're using it as a discriminator.

Reply:
done, also added OnPremise value because one valued enum is not included in the .Net ctors and creates a breaking change when the new value is added.
we will add the relevant OnPremiseResourceDetails before the completion of this PR, but we still need to work on them